content/post/language_migrations.md

```diff
@@ -0,0 +1,512 @@
+
+<div class="grid2x">
+<div>
+<div>
+<img src="/post/language_migrations/sum_matrix_wdiag_22lang.svg">
+<p align="center" class="dt"><pre><code class="hljs python">source{d}'s flow transition matrix</code></pre></p>
+</div>
+</div>
+<div>
+<div>
+<img src="/post/language_migrations/erik_red.png">
+<p align="center" class="dt"><pre><code class="hljs python">Erik's contingency table</code></pre></p>
+</div>
+</div>
+</div>
+
+The first thing to note here is that comparing to Erik's table we have the transition matrix's diagonal. We will see later how we could take advantage of it. However, although the datasets we used are completely different than Erik's, we can notice many relevant similarities and perceive the same kind of language profile.
+
+## GitHub "LanguageRank"
+
+Since we have our own flow matrix, we want to know which languages are the most and the less popular. The best way to do so, is to compute [centrality measures](https://en.wikipedia.org/wiki/Centrality) thanks to eigenvector calculations. Indeed those measures would convey the relative popularity of languages in the sense that people coding in a language would be more or less likely to switch for another one. Let's now details the main ideas of this approach. If you need further explanations, I invite you to read [Vadim Markovtsev](https://github.com/vmarkovtsev?tab=repositories)'s PageRank analysis in his blog post about the [GitHub Contributions Graph](https://blog.sourced.tech/post/handshakes_pagerank/).
+
+1. Our flow matrix contains strictly positive elements, which is a sufficient condition to make it [irreducible](https://en.wikipedia.org/wiki/Irreducibility_(mathematics)) ; from any language there is a way to reach all other ones. Thus, according to [Perron–Frobenius theorem](https://en.wikipedia.org/wiki/Perron%E2%80%93Frobenius_theorem), we are looking for the greatest eigenvalue and the corresponding eigenvector.
+
+2. To find such the dominant eigenvector, we can use the [Power Iteration](https://en.wikipedia.org/wiki/Power_iteration) algorithm. Nevertheless, in addition to be irreducible, the matrix needs to be stochastic and aperiodic.
+
+3. We make our flow matrix become [stochastic](https://en.wikipedia.org/wiki/Stochastic_matrix) by normalizing rows after removing the main diagonal. The greatest eigenvalue is now equal to 1.
+
+4. At last, to make our matrix [aperiodic](https://en.wikipedia.org/wiki/Markov_chain) and well conditioned anyway, there is a famous trick introduced by Larry and Sergey in 1998. It is well explained in [Stanford’s CS246](http://snap.stanford.edu/class/cs246-2013/slides/09-pagerank.pdf) but to make it short, it mainly consists in updating our flow matrix with the following formula:
```

Comment:
~At last~ Finally ... ~anyway~

... too many "make" ... please replace where possible

mainly consists **of** ... using the following formula